app/jobs/notify_followers_of_new_post_job.rb

```diff
@@ -44,21 +44,21 @@ def notify_of_post_joining(post, new_user)
       next if already_notified_about?(post, user)
       message = "#{new_user.username} has just joined the post entitled #{post.subject} with "
       message += post.joined_authors.where.not(id: new_user.id).pluck(:username).join(', ')
-      message += ". #{view_post(post.id)}"
+      message += ". #{ScrapePostJob.view_post(post.id)}"
       Message.send_site_message(user.id, subject, message)
     end
   end
 
   def already_notified_about?(post, user)
-    Message.where(recipient: user, sender_id: 0).find_each do |notification|
-      return true if notification.message.include?(view_post(post.id))
-    end
-    false
+    self.class.notification_about(post, user).present?
   end
 
-  def view_post(post_id)
-    host = ENV['DOMAIN_NAME'] || 'localhost:3000'
-    url = Rails.application.routes.url_helpers.post_url(post_id, host: host, protocol: 'https')
-    "<a href='#{url}'>View it here</a>."
```

Comment:
Was this duplicated on ScrapePostJob? I'm not sure that's the best place to centralize if it was but otherwise this looks fine if so.

Reply:
Yeah, it was duplicated. Didn't care too much about where to put it because Someday Notifications and we won't need it.